### ISSUE TITLE
[TV4] HTTP Error 503: Service Unavailable

### ISSUE THREAD

**Issue by TxI5** [yt-dlp/yt-dlp#5535]
### DO NOT REMOVE OR SKIP THE ISSUE TEMPLATE

- [X] I understand that I will be **blocked** if I remove or skip any mandatory\* field

### Checklist

- [X] I'm reporting a broken site
- [X] I've verified that I'm running yt-dlp version **2022.11.11** ([update instructions](https://github.com/yt-dlp/yt-dlp#update)) or later (specify commit)
- [X] I've checked that all provided URLs are playable in a browser with the same IP and same login details
- [X] I've checked that all URLs and arguments with special characters are [properly quoted or escaped](https://github.com/yt-dlp/yt-dlp/wiki/FAQ#video-url-contains-an-ampersand--and-im-getting-some-strange-output-1-2839-or-v-is-not-recognized-as-an-internal-or-external-command)
- [X] I've searched the [bugtracker](https://github.com/yt-dlp/yt-dlp/issues?q=) for similar issues **including closed ones**. DO NOT post duplicates
- [X] I've read the [guidelines for opening an issue](https://github.com/yt-dlp/yt-dlp/blob/master/CONTRIBUTING.md#opening-an-issue)
- [X] I've read about [sharing account credentials](https://github.com/yt-dlp/yt-dlp/blob/master/CONTRIBUTING.md#are-you-willing-to-share-account-details-if-needed) and I'm willing to share it if required

### Region

Sweden

### Provide a description that is worded well enough to be understood

www.tv4play.se not working:
Unable to download JSON metadata: HTTP Error 503: Service Unavailable

### Provide verbose output that clearly demonstrates the problem

- [X] Run **your** yt-dlp command with **-vU** flag added (`yt-dlp -vU <your command line>`)
- [X] Copy the WHOLE output (starting with `[debug] Command-line config`) and insert it below

### Complete Verbose Output

```shell
[debug] Command-line config: ['https://www.tv4play.se/program/next-level-chef-us/del-5-s%C3%A4song-1/13772514', '-vU']
[debug] Encodings: locale UTF-8, fs utf-8, pref UTF-8, out utf-8, error utf-8, screen utf-8
[debug] yt-dlp version 2022.11.11 [8b64402] (pip)
[debug] Python 3.10.6 (CPython x86_64 64bit) - Linux-5.15.0-52-generic-x86_64-with-glibc2.35 (OpenSSL 3.0.2 15 Mar 2022, glibc 2.35)
[debug] exe versions: ffmpeg 4.4.2 (setts), ffprobe 4.4.2, rtmpdump 2.4
[debug] Optional libraries: Cryptodome-3.11.0, brotli-1.0.9, certifi-2020.06.20, mutagen-1.46.0, pyxattr-0.7.2, secretstorage-3.3.1, sqlite3-2.6.0, websockets-10.4
[debug] Proxy map: {}
[debug] Loaded 1723 extractors
[debug] Fetching release info: https://api.github.com/repos/yt-dlp/yt-dlp/releases/latest
Latest version: 2022.11.11, Current version: 2022.11.11
yt-dlp is up to date (2022.11.11)
[debug] Using fake IP 78.67.182.65 (SE) as X-Forwarded-For
[debug] [TV4] Extracting URL: https://www.tv4play.se/program/next-level-chef-us/del-5-s%C3%A4song-1/13772514
[TV4] 13772514: Downloading video info JSON
ERROR: [TV4] 13772514: Unable to download JSON metadata: HTTP Error 503: Service Unavailable (caused by <HTTPError 503: 'Service Unavailable'>); please report this issue on  https://github.com/yt-dlp/yt-dlp/issues?q= , filling out the appropriate issue template. Confirm you are on the latest version using  yt-dlp -U
  File "/home/user/.local/lib/python3.10/site-packages/yt_dlp/extractor/common.py", line 674, in extract
    ie_result = self._real_extract(url)
  File "/home/user/.local/lib/python3.10/site-packages/yt_dlp/extractor/tv4.py", line 75, in _real_extract
    info = self._download_json(
  File "/home/user/.local/lib/python3.10/site-packages/yt_dlp/extractor/common.py", line 1034, in download_content
    res = getattr(self, download_handle.__name__)(url_or_request, video_id, **kwargs)
  File "/home/user/.local/lib/python3.10/site-packages/yt_dlp/extractor/common.py", line 998, in download_handle
    res = self._download_webpage_handle(
  File "/home/user/.local/lib/python3.10/site-packages/yt_dlp/extractor/common.py", line 868, in _download_webpage_handle
    urlh = self._request_webpage(url_or_request, video_id, note, errnote, fatal, data=data, headers=headers, query=query, expected_status=expected_status)
  File "/home/user/.local/lib/python3.10/site-packages/yt_dlp/extractor/common.py", line 825, in _request_webpage
    raise ExtractorError(errmsg, cause=err)

  File "/home/user/.local/lib/python3.10/site-packages/yt_dlp/extractor/common.py", line 807, in _request_webpage
    return self._downloader.urlopen(self._create_request(url_or_request, data, headers, query))
  File "/home/user/.local/lib/python3.10/site-packages/yt_dlp/YoutubeDL.py", line 3692, in urlopen
    return self._opener.open(req, timeout=self._socket_timeout)
  File "/usr/lib/python3.10/urllib/request.py", line 525, in open
    response = meth(req, response)
  File "/usr/lib/python3.10/urllib/request.py", line 634, in http_response
    response = self.parent.error(
  File "/usr/lib/python3.10/urllib/request.py", line 563, in error
    return self._call_chain(*args)
  File "/usr/lib/python3.10/urllib/request.py", line 496, in _call_chain
    result = func(*args)
  File "/usr/lib/python3.10/urllib/request.py", line 643, in http_error_default
    raise HTTPError(req.full_url, code, msg, hdrs, fp)
urllib.error.HTTPError: HTTP Error 503: Service Unavailable
```

**Comment by akvadrat** [comment]
I got the same issue. My guess it has to do with the fact that you have to login using a free account on tv4play now. This seems to have happened just recently.

![image](https://user-images.githubusercontent.com/10449598/201760420-1df3b169-3473-4c30-928b-9f468b48169a.png)

**Comment by pukkandan** [comment]
I don't get any login prompt, but the provided link appears to be DRM protected

**Comment by akvadrat** [comment]
Should be the same issue as this: https://github.com/spaam/svtplay-dl/issues/1509
Fix: https://github.com/spaam/svtplay-dl/commit/8950be8eec917b3bf206c3d32d6df2a6fe4797be

**Comment by pukkandan** [comment]
> And with my little knowledge I don't think it is DRM-protected.

Why do you say so?

![image](https://user-images.githubusercontent.com/6623388/202781346-bc8b6fb1-9955-4ada-90ab-53be43332922.png)

**Comment by Andber8** [comment]
Since It is possible to download with https://github.com/spaam/svtplay-dl/ with the latest fix and that program does not "decode" DRM as far as I know.
But maybe I have misunderstood that.

**Comment by pukkandan** [comment]
> Fix: [spaam/svtplay-dl@8950be8](https://github.com/spaam/svtplay-dl/commit/8950be8eec917b3bf206c3d32d6df2a6fe4797be)

Ah, my bad. I did not notice this part in previous comment 😓

Does this work with cookies?

```diff
diff --git a/yt_dlp/extractor/tv4.py b/yt_dlp/extractor/tv4.py
index 1378a6f57..22ee7e2f3 100644
--- a/yt_dlp/extractor/tv4.py
+++ b/yt_dlp/extractor/tv4.py
@@ -73,12 +73,14 @@ def _real_extract(self, url):
         video_id = self._match_id(url)

         info = self._download_json(
-            'https://playback-api.b17g.net/asset/%s' % video_id,
+            f'https://playback2.a2d.tv/play/{video_id}',
             video_id, 'Downloading video info JSON', query={
                 'service': 'tv4',
                 'device': 'browser',
                 'protocol': 'hls,dash',
                 'drm': 'widevine',
+                'browser': 'GoogleChrome',
+                'capabilities': 'live-drm-adstitch-2,expired_assets',
             })['metadata']

         title = info['title']
```

**Comment by TxI5** [comment]
It does not work for me with fix, with or without cookies.
Latest svtplay-dl does not work either.

502 Bad Gateway. Does it work for anyone else?

**Comment by TxI5** [comment]
So it might work now. The 502 was for the manifest_url still pointing to the b17g domain. 

> > Fix: [spaam/svtplay-dl@8950be8](https://github.com/spaam/svtplay-dl/commit/8950be8eec917b3bf206c3d32d6df2a6fe4797be)
> 
> Ah, my bad. I did not notice this part in previous comment 😓
> 
> Does this work with cookies?
> 
> ```diff
> diff --git a/yt_dlp/extractor/tv4.py b/yt_dlp/extractor/tv4.py
> index 1378a6f57..22ee7e2f3 100644
> --- a/yt_dlp/extractor/tv4.py
> +++ b/yt_dlp/extractor/tv4.py
> @@ -73,12 +73,14 @@ def _real_extract(self, url):
>          video_id = self._match_id(url)
> 
>          info = self._download_json(
> -            'https://playback-api.b17g.net/asset/%s' % video_id,
> +            f'https://playback2.a2d.tv/play/{video_id}',
>              video_id, 'Downloading video info JSON', query={
>                  'service': 'tv4',
>                  'device': 'browser',
>                  'protocol': 'hls,dash',
>                  'drm': 'widevine',
> +                'browser': 'GoogleChrome',
> +                'capabilities': 'live-drm-adstitch-2,expired_assets',
>              })['metadata']
> 
>          title = info['title']
> ```
> 
> PS: Related [spaam/svtplay-dl#1509](https://github.com/spaam/svtplay-dl/issues/1509)


This fix only changes the info_url. The manifest_url also has the be changed to 'playback2.a2d.tv/play/'. The info_url might be 'playback2.a2d.tv/assets' actually?
